src/modules/service-provider/api/provider.controller.ts

```diff
@@ -48,6 +53,36 @@ export class ProviderController {
         return response;
     }
 
+    @Get()
+    @ApiOkResponse({
+        description: 'The service-providers were successfully returned.',
+        type: [ServiceProviderResponse],
+    })
+    @ApiUnauthorizedResponse({ description: 'Not authorized to get available service providers.' })
+    @ApiForbiddenResponse({ description: 'Insufficient permissions to get service-providers.' })
+    @ApiInternalServerErrorResponse({ description: 'Internal server error while getting all service-providers.' })
+    public async getAvailableServiceProviders(
+        @Permissions() permissions: PersonPermissions,
+    ): Promise<ServiceProviderResponse[]> {
+        const roleIds: string[] = await permissions.getRoleIds();
+        const serviceProviders: ServiceProvider<true>[] = [];
+        for (const roleId of roleIds) {
+            const rolle: Option<Rolle<true>> = await this.rolleRepo.findById(roleId);
+            if (rolle) {
+                for (const serviceProviderId of rolle.serviceProviderIds) {
+                    const serviceProvider: Option<ServiceProvider<true>> =
+                        await this.serviceProviderRepo.findById(serviceProviderId);
+                    if (serviceProvider) {
+                        serviceProviders.push(serviceProvider);
+                    }
+                }
+            }
+        }
```

Comment:
If I read this correctly, this could potentially include the same ServiceProvider multiple times, if a user has multiple roles that have access to the same ServiceProviders

Reply:
yes, will add a check for that